### PR TITLE
lax, moves list of environments to skip glencoe caching into pillar

### PR DESCRIPTION
**Pull request by lsh-0**

### CODE REVIEW

File: pillar/lax-public.sls [hunk 1]
```diff
@@ -70,6 +70,12 @@ elife:
                 service_template: bot-lax-adaptor-service
                 num_processes: 1
 
+    glencoe:
+        skip_requests_cache_by_env:
```
Comment (giorgiosironi) [inline pillar/lax-public.sls]: just needs to be a single boolean value that changes by environment? :thinking: 
Like e.g. `pillar.journal.submit_url` as an example?

Comment (lsh-0) [reply pillar/lax-public.sls]: I'll take a second look

Comment (lsh-0) [reply pillar/lax-public.sls]: > just needs to be a single boolean value that changes by environment? 

I've taken a look and what you're suggesting and I think my approach is better. In this case we have one file to change where a value is set. In your approach (correct me if I'm wrong) there are N files to modify, one for each environment. That would seem to encourage file sprawl and obfuscation at the expense of simpler formulas (I think? is the intent to keep environment wrangling out of applications?). Looking at some of the `environment-whatever-public.sls` files, it doesn't appear to have stopped the use of value templating or using environment names in pillar values.

I'm not on a crusade, I think we're just merrily dolling out all the rope we need to hang ourselves with. I can understand partitioning public/private pillar data, and perhaps that was the top of the slippery slope, but not further partitioning by environment (and do we have individual project instances in there? `environment-title-length-...` ?). We should have found some way to push per-environment stuff back into the public/private project pillar data. Maybe this: https://docs.saltstack.com/en/latest/ref/states/top.html#multiple-environments . I never had the time to bend my head around it though.

I'll change things so this pillar value is changed via the environment files

**Comment by giorgiosironi** [comment]
> is the intent to keep environment wrangling out of applications?

The intent is to be able to create new environments without touching applications, which could also be achieved by listing environment names in a pillar variable and checking `pillar.elife.env` against them. I didn't necessarily see as a simplification though, because in [this version](https://github.com/elifesciences/lax-formula/pull/37/commits/574bcf2fed525f4684d3505bb7ef1d47e740378b#diff-3542b2ecf7bb0000d2f13b09f8d81b58R29) the formula acquires logic (`not in` is essentially an if/else to produce the two possible boolean values) whereas in the current version the only logic should be the merging of pillars; it does privilege environment isolation and decoupling to the cohesion of being able to compare what environment do in a single place (like that `lax.sls` file) however.

> Maybe this: https://docs.saltstack.com/en/latest/ref/states/top.html#multiple-environments . I never had the time to bend my head around it though.

When I originally looked into it, it seemed heavyweight as it required to copy identical files across the subfolders that represent environments.

**Pull request merged**